src/OrchardCore/OrchardCore/Environment/Shell/Builders/ShellContainerFactory.cs

```diff
@@ -159,5 +153,23 @@ private void EnsureApplicationFeature()
                 }
             }
         }
+
+        internal Type GetImplementationType(ServiceDescriptor descriptor)
+        {
+            if (descriptor.ImplementationType != null)
```

Comment:
(descriptor.ImplementationType is object)  is faster and better IL :)

Reply:
https://twitter.com/FransBouma/status/1115201371182071808

Reply:
Really? Thanks for the info, will do

Reply:
hehehe